Straight_skeleton_2/examples/Straight_skeleton_2/include/CGAL/input_helpers.h

```diff
@@ -0,0 +1,256 @@
+bool read_segment_speeds(const char* filename,
+                        std::vector<std::vector<FT> >& weights)
+{
+  std::ifstream in(filename);
+  if(!in)
+  {
+    std::cerr << "Error: Could not read " << filename << std::endl;
+    return false;
+  }
+
+  std::vector<FT> border_weights;
+
+  std::string line;
+  while(getline(in, line))
+  {
+    if(line.empty())
+    {
+      weights.push_back(border_weights);
+      border_weights.clear();
+    }
+
+    std::istringstream iss(line);
+    double w;
+    if(iss >> w)
+      border_weights.push_back(w);
+  }
+
+  // in case the last line is not empty
+  if(!border_weights.empty())
+    weights.push_back(border_weights);
+
+  return true;
+}
+
+// create a random 10-gon in a square
+template <typename PolygonWithHoles>
+PolygonWithHoles generate_random_polygon(const int seed)
+{
+  using Polygon_2 = typename PolygonWithHoles::Polygon_2;
+  using Point_2 = typename Polygon_2::Point_2;
+
+  typedef CGAL::Random_points_in_square_2<Point_2> Point_generator;
+
+  CGAL::Random rnd(seed);
+
+  Polygon_2 poly;
+  CGAL::random_polygon_2(10, std::back_inserter(poly), Point_generator(0.5, rnd));
+  return PolygonWithHoles{poly};
+}
+
+template <typename PolygonWithHoles, typename FT>
+void generate_random_weights(const PolygonWithHoles& p,
+                             const double min_weight,
+                             const double max_weight,
+                             const int seed,
+                             std::vector<std::vector<FT> >& speeds)
+{
+  using Polygon_2 = typename PolygonWithHoles::Polygon_2;
+  using Point_2 = typename Polygon_2::Point_2;
+  using K = typename CGAL::Kernel_traits<Point_2>::type;
+  using Segment_2 = typename K::Segment_2;
+
+  CGAL::Random rnd(seed);
+  std::cout << "Seed is " << rnd.get_seed() << std::endl;
+
+  CGAL_assertion(max_weight > 1);
+
+  auto prev = [](const auto& it, const auto& container)
+  {
+    return it == container.begin() ? std::prev(container.end()) : std::prev(it);
+  };
+
+  auto next = [](const auto& it, const auto& container)
+  {
+    return (it == std::prev(container.end())) ? container.begin() : std::next(it);
+  };
+
+  auto generate_range_weights = [prev, next, min_weight, max_weight, &rnd](const auto& c)
+  {
+    using Container = typename std::remove_reference<decltype(c)>::type;
+    using Iterator = typename Container::const_iterator;
+
+    std::map<Iterator, std::size_t /*rnd weight*/> weight;
+
+    // start somewhere not collinear
+    Iterator start_it;
+    for(Iterator it=c.begin(); it<c.end(); ++it)
+    {
+      // the edge is [prev_1 ; it], check for collinearity with the previous edge [prev_2; prev_1]
+      auto prev_2 = prev(prev(it, c), c);
+      auto prev_1 = prev(it, c);
+      Segment_2 s0 {*prev_2, *prev_1}, s1 {*prev_1, *it};
+      if(!CGAL::CGAL_SS_i::are_edges_orderly_collinear(s0, s1))
+        start_it = it;
+    }
+
+    CGAL_assertion(start_it != Iterator()); // all collinear is impossible
```

Comment:
VC does not lilke this as it checks that the iterators come from the same container.  Can we  compare with `c.end()` @MaelRL ?